lib/grpc/adapter/client.ex

```diff
@@ -0,0 +1,27 @@
+defmodule GRPC.ClientAdapter do
+  @moduledoc false
+
+  alias GRPC.Client.Stream
+  alias GRPC.Channel
+
+  @type opts :: any
+
+  @type channel :: Channel.t()
+  @callback connect(channel, opts) :: {:ok, channel} | {:error, any}
+  
+  @callback disconnect(channel) :: {:ok, channel} | {:error, any}
+
+  @type stream :: Stream.t()
```

Comment:
I would use `Stream.t()` directly instead of adding this proxy type. Is there any reason for this path?

Reply:
For clarity, I would also move all `@type` definitions to the top of the module

Reply:
Using the in-line type is better for readability:

```elixir
# Module defs
defmodule T do

@type t :: integer

end

defmodule M do
  @type custom :: T.t()
  @spec f(x :: T.t(), y :: custom) :: T.t()
  def f(x, y), do: x + y
end

# iex calls
iex(1)> t M  
@type custom() :: T.t()

iex(2)> h M.f

                                  def f(x, y)                                   

  @spec f(x :: T.t(), y :: custom()) :: T.t()
```

When exploring the API via IEx, the proxy type will force the user to call `t module` to get the typedefs and discover it is indeed a proxy